nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CalculateAttributeHash.java

```diff
@@ -0,0 +1,199 @@
+
+    private final AtomicReference<Map<String, String>> attributeToGenerateNameMapRef = new AtomicReference<>(Collections.emptyMap());
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_SUCCESS);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(HASH_ALGORITHM);
+        this.properties = Collections.unmodifiableList(properties);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName).addValidator(StandardValidators.NON_BLANK_VALIDATOR).build();
+    }
+
+    @Override
+    public void onPropertyModified(final PropertyDescriptor descriptor, final String oldValue, final String newValue) {
+        if (descriptor.isRequired()) {
+            return;
+        }
+
+        final Map<String, String> attributeToGeneratedNameMap = new HashMap<>(attributeToGenerateNameMapRef.get());
+        if (newValue == null) {
+            attributeToGeneratedNameMap.remove(descriptor.getName());
+        } else {
+            attributeToGeneratedNameMap.put(descriptor.getName(), newValue);
+        }
+
+        attributeToGenerateNameMapRef.set(Collections.unmodifiableMap(attributeToGeneratedNameMap));
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final Map<String, String> attributeToGeneratedNameMap = attributeToGenerateNameMapRef.get();
+        final ComponentLog logger = getLogger();
+
+        final SortedMap<String, String> attributes = getRelevantAttributes(flowFile, attributeToGeneratedNameMap);
+        if (attributes.size() != attributeToGeneratedNameMap.size()) {
+            final Set<String> wantedKeys = attributeToGeneratedNameMap.keySet();
+            final Set<String> foundKeys = attributes.keySet();
+            final StringBuilder missingKeys = new StringBuilder();
+            for (final String wantedKey : wantedKeys) {
+                if (!foundKeys.contains(wantedKey)) {
+                    missingKeys.append(wantedKey).append(" ");
+                }
+            }
+            logger.info("routing {} to 'failure' because of missing attributes: {}", new Object[]{flowFile, missingKeys.toString()});
+            session.transfer(flowFile, REL_FAILURE);
```

Comment:
This seems pretty inflexible. If this exists at the intersection between multiple flows, it's possible the user could want to have dozens of possible hashing jobs, but each flow would have only 25-30% of the wanted keys.

Reply:
This is per the existing behavior of the HashAttribute processor.  I will change it to be  governed by a property